gpio_controllers/CMakeLists.txt

```
@@ -0,0 +1,113 @@
+cmake_minimum_required(VERSION 3.8)
+project(gpio_controllers)
+
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
```

```suggestion
  add_compile_options(-Wall -Wextra -Wpedantic -Werror=conversion -Werror=unused-but-set-variable -Werror=return-type -Werror=shadow -Werror=format -Werror=range-loop-construct)
```

Can we test with these parameters, if the code is compiling or not?
@christophfroehlich @Wiktor-99

---

These are the standard flags we tend to have across all the packages